crates/trigger/src/loader.rs

```diff
@@ -13,16 +11,22 @@ use tokio::fs;
 
 use spin_common::{ui::quoted_path, url::parse_file_url};
 
+/// Loader for the Spin runtime
 pub struct TriggerLoader {
+    /// Working directory where files for mounting exist.
     working_dir: PathBuf,
+    /// Set the static assets of the components in the temporary directory as writable.
     allow_transient_write: bool,
+    /// All components have been ahead of time (AOT) compiled (to cwasm) and should be loaded through deserialization.
+    aot: bool,
```

Comment:
The boolean here is hard to interpret at point of use (e.g. in the `TriggerLoader` constructor call or when checking `self.aot`).  (Particularly since `allow_transient_write` is also a boolean so you end up with a sequence of booleans in the constructor call and what do they mean.)  Consider an enum such as `CompilationStatus::AllAot` or `Aot::AllAotCompiled`.

(and yes, `allow_transient_writes` should have been an enum for the same reasons, but that's out of scope!)